src/dbnode/client/session.go

```diff
@@ -1557,6 +1559,7 @@ func (s *session) fetchIDsAttempt(
 				slicesIter := s.pools.readerSliceOfSlicesIterator.Get()
 				slicesIter.Reset(result.([]*rpc.Segments))
 				multiIter := s.pools.multiReaderIterator.Get()
+				multiIter.SetSchema(nsCtx.Schema)
```

Comment:
Passing `schema` could be moved to an arg of `Reset(...)` here too.

Reply:
ditto